### PR TITLE
feat: add commitment param to JSON RPC API requests

### PR DESCRIPTION
**Pull request by jstarry**
Fixes: https://github.com/solana-labs/solana-web3.js/issues/546

Other related changes:
* Allow specifying a default commitment level for a `Connection` instance
* Add new `sendAndConfirmRecentTransaction` convenience method
* Fix broken examples
* Fix live tests
* Add test for max commitment

### CODE REVIEW

**Comment by jstarry** [comment]
Tests are still failing when `to` is unsigned, I think those can be fixed in a follow up.

File: src/connection.js [hunk 1]
```diff
@@ -18,6 +18,27 @@ import type {TransactionSignature} from './transaction';
...
+ * @typedef {'recent'} RECENT_COMMITMENT
+ */
+export type RECENT_COMMITMENT = 'recent';
+
+/**
+ * The level of commitment desired when querying state
+ *
+ * @typedef {MAX_COMMITMENT | RECENT_COMMITMENT} Commitment
+ */
+export type Commitment = MAX_COMMITMENT | RECENT_COMMITMENT;
```
Comment (mvines) [inline src/connection.js]: I prefer just a simple: 
```js
export type Commitment = 'max' | 'recent';
```

flow will yell if any other strings are used

Comment (jstarry) [reply src/connection.js]: yeah much better!

**Pull request merged**